### PR TITLE
Jenkins operator

### PR DESCRIPTION
**Pull request by JonRudy**
Creating node module to run terraform commands inside jenkins operator / created pvc template and updated jenkins deployment to include a pvc volume

### CODE REVIEW

File: charts/operator-toolchain/templates/deployment.yaml [hunk 1]
```diff
@@ -29,20 +29,29 @@ spec:
         - name: operator-{{ $name}}
           image: {{ $operator.image }}
           imagePullPolicy: {{ $operator.pullPolicy }}
-          {{- if $operator.configSecret }}
           env:
+          {{- if $operator.configSecret }}
```
Comment (stein321) [inline charts/operator-toolchain/templates/deployment.yaml]: so if there's a config secret we set the config file as an env variable?

File: operator-jenkins/Dockerfile [hunk 2]
```diff
@@ -12,9 +13,21 @@ FROM node:10.15-alpine as prod
 WORKDIR /app
 ENV PORT 3000
 
+RUN apk add --no-cache \
+    curl \
+    unzip
+
+ENV TERRAFORM_VERSION 0.12.0
+RUN curl -f -Lo terraform.zip https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
```
Comment (stein321) [inline operator-jenkins/Dockerfile]: how is terraform going to talk to the cluster?

File: charts/operator-toolchain/templates/deployment.yaml (hunk 1 above)
Comment (cplee) [reply charts/operator-toolchain/templates/deployment.yaml]: Yup....line 34

File: operator-jenkins/Dockerfile (hunk 2 above)
Comment (cplee) [reply operator-jenkins/Dockerfile]: There will be a service account attached to the pod with a kubeconfig available to authenticate with

**Pull request merged**